src/GraphQL/Resolvers/MethodResolver.cs

```diff
@@ -42,11 +47,9 @@ public MethodResolver(MethodInfo methodInfo, List<LambdaExpression> methodArgume
                 Expression.Call(
                     methodInfo.IsStatic
                         ? null
-                        : Expression.Convert(
-                            Expression.MakeMemberAccess(
-                                resolveFieldContextParameter,
-                                _resolveFieldContextSourceParameter),
-                            methodInfo.DeclaringType!),
+                        : sourceExpression.Body.Replace(
+                            sourceExpression.Parameters[0],
+                            resolveFieldContextParameter),
                     methodInfo,
```

Comment:
So NRE in case of null `Source` happens here, right? On `methodInfo` access. If so maybe change 
` private static readonly Expression<Func<IResolveFieldContext, TSourceType>> _sourceExpression = context => (TSourceType)context.Source!;`
to
` private static readonly Expression<Func<IResolveFieldContext, TSourceType>> _sourceExpression = context => (TSourceType)context.Source ?? throw new InvalidOperationException("some informative message");
`
?

Reply:
Changed as discussed elsewhere.